.idea/libraries/Dart_practice/chapter_four.dart

```diff
@@ -0,0 +1,233 @@
+//    if (firstName == 'Bob') {
+//        const lastName = 'Smith';
+//    } else if (firstName == 'Ray') {
+//        const lastName = 'Wenderlich';
+//    }
+//    final fullName = firstName + ' ' + lastName;”
+
+
+  const firstName = 'Bob';
+  var lastName = "";
+  if (firstName == 'Bob') {
+    lastName = 'Smith';
+  } else if (firstName == 'Ray') {
+    lastName = 'Wenderlich';
+  }
+  final fullName = firstName + ' ' + lastName;
+  print("Full name is $fullName");
+  print("---------------------------");
+
+  print("Challenge 2: Boolean challenge!");
+
+  // 1.  “In each of the following statements, what is the value of the Boolean expression?”
+
+  const firstStatement = true && true;
+  print("Value is $firstStatement.");
+  const secondStament = false || false;
+  print("Value is $secondStament.");
+  const thirdStatement = (true && 1 != 2) || (4 > 3 && 100 < 1);
+  print("Value is $thirdStatement.");
+  const forthStatement = ((10 / 2) > 3) && ((10 % 2) == 0);
+  print("Value is $forthStatement");
+  print("---------------------------");
+
+  print("Challenge 3: Next power of two!");
+
+  // 1. “Given a number, determine the next power of two above or equal to that number.”
+  // 2. “Powers of two are the numbers in the sequence of 2¹, 2², 2³, and so on.”
+  // 3. “You may also recognize the series as 1, 2, 4, 8, 16, 32, 64...”
+
+  const n = 12;
+  for (var i = 1; i < 10; i++) {
+    var powerOfTwo = pow(2, i);
+    if (powerOfTwo >= n) {
+      print(powerOfTwo);
+      break;
+    }
+  }
+  print("---------------------------");
+
+  print("Challenge 4: Fibonacci number!");
+
+  // 1. “Calculate the nth Fibonacci number.
+  // 2. "The Fibonacci sequence starts with 1, then 1 again, and then all subsequent numbers in the sequence are simply the previous two values in the sequence added together (1, 1, 2, 3, 5, 8...)”
+
+  // Fibonacci sequence algo (simple) >>> calculate the next number by adding the current number to the old number;
+
+  var num = 9;
+  var a = 1;
+  var b = 0;
+  var temp = 0;
+
+  for (var i = 0; i < 30; i++) {
+    temp = a;
+    a = a + b;
+    b = temp;
+    if ( i == num) {
+      print("The $num-th number in the Fibonacci sequence is $b.");
+      break;
+    }
+  }
+  print("---------------------------");
+
+  print("Challenge 4: How many times?");
+
+  // 1. In the following for loop, what will be the value of sum, and how many iterations will happen?
+  // var sum = 0;
+  // for (var i = 0; i <= 5; i++) {
+  //   sum += i;
+  //   }
+
+  var sum = 0;
+  var iterations = 0;
+  for (var i = 0; i <= 5; i++) {
+    sum += i;
+    print("Sum is $sum.");
+    iterations = i + 1;
+    print("Iteration is $iterations.");
+  }
+  print("The sum is $sum, and the number of iterations is $iterations.");
+  print("---------------------------");
+
+  print("Challenge 4: The final countdown");
+
+  // 1. Print a countdown from 10 to 0.
+
+  for (var i = 0; i <= 10; i++) {
+    var countdown = 10 - i;
```

Comment:
@octavianChiriches what if you start the loop at 10, exit at 0 and decrement (i--). 
Try it with a while loop too.